applications/main/bad_usb/helpers/ducky_script_commands.c

```diff
@@ -124,32 +125,64 @@ static int32_t ducky_fnc_altstring(BadUsbScript* bad_usb, const char* line, int3
 
 static int32_t ducky_fnc_hold(BadUsbScript* bad_usb, const char* line, int32_t param) {
     UNUSED(param);
+    bool isMouse = false;
 
     line = &line[ducky_get_command_len(line) + 1];
+
+    // Handle Keyboard keys here
     uint16_t key = ducky_get_keycode(bad_usb, line, true);
     if(key == HID_KEYBOARD_NONE) {
         return ducky_error(bad_usb, "No keycode defined for %s", line);
     }
+
+    // Handle Mouse Keys here
+    key = ducky_get_mouse_keycode_by_name(line);
+    if(key != HID_MOUSE_INVALID) {
+        isMouse = true;
+    }
+
     bad_usb->key_hold_nb++;
     if(bad_usb->key_hold_nb > (HID_KB_MAX_KEYS - 1)) {
         return ducky_error(bad_usb, "Too many keys are hold");
     }
+
+    if(isMouse) {
+        bad_usb->hid->mouse_press(bad_usb->hid_inst, key);
+        return 0;
+    }
+
     bad_usb->hid->kb_press(bad_usb->hid_inst, key);
     return 0;
 }
 
 static int32_t ducky_fnc_release(BadUsbScript* bad_usb, const char* line, int32_t param) {
     UNUSED(param);
+    bool isMouse = false;
 
     line = &line[ducky_get_command_len(line) + 1];
+
+    //Handle Keyboard Keys here
     uint16_t key = ducky_get_keycode(bad_usb, line, true);
     if(key == HID_KEYBOARD_NONE) {
         return ducky_error(bad_usb, "No keycode defined for %s", line);
     }
+
+    // Handle Mouse Keys here
+    key = ducky_get_mouse_keycode_by_name(line);
```

Comment:
I understand that badusb code is not the best example, but I got feeling that what you doing here makes it even more difficult to read. 

Do you mind to try to refactor keycode parsing and handling in more explicit way? In a way where if you started mouse branch then you do mouse branch and there is no places where keyboard and mouse collide?